### ISSUE TITLE
Add integration tests for interoperability testing with other extensions libraries

### ISSUE THREAD

**Issue by sangonzal** [AzureAD/microsoft-authentication-extensions-for-python#82]
We should have automated tests for validating interoperability in between .NET, Java, and Node extensions libraries. The tests can live in one repository, as the cache interoperability tests do.

**Comment by rayluo** [comment]
What kind of test cases do you have in mind?

> tests for validating **interoperability in between ... extensions libraries** ... as the **cache interoperability tests** do

Not sure how other MSAL EX are implemented. MSAL EX for Python does not control the cache schema, so, the two things that I highlighted above, are the same thing in Python side, thus already covered by the "cache interoperability tests" that you mentioned (although we could always add more test cases into it, if that's what you mean).

MSAL EX's job is encryption and locking. It is true that we do not currently have interoperability test cases for these 2 areas.

**Comment by sangonzal** [comment]
Yes, that's right, encryption and locking are the two areas of interest here. I don't think encryption necessarily needs to be tested, as the libraries use all of the same system APIs (DPAPI on Windows for all libraries, Keychain on Mac, Keyring on Linux). What should be tested though is file locking in between libraries. Specifically that filelocks are respected across libraries (if a Python process has a lock, the Java process should hold off until the lock is released, and vice versa). This could be scaled to multiple processes as a stress test, as some of the unit tests already do in Python. These tests could actually be added to the cache interoperability test framework that already exists, in which we could test both schema and the extensions - the whole E2E customer experience.
